Lombiq.Tests.UI/Services/UITestExecutionSession.cs

```diff
@@ -223,18 +235,28 @@ private async Task CreateFailureDumpAsync(Exception ex, string dumpRootPath, int
 
                 if (_context == null) return;
 
-                // Saving the screenshot and HTML output should be as early after the test fail as possible so they show
-                // an accurate state. Otherwise, e.g. the UI can change, resources can load in the meantime.
-                if (_dumpConfiguration.CaptureScreenshot)
+                // Saving the failure screenshot and HTML output should be as early after the test fail as possible so
+                // they show an accurate state. Otherwise, e.g. the UI can change, resources can load in the meantime.
+                if (_dumpConfiguration.CaptureScreenshots)
                 {
                     // Only PNG is supported on .NET Core.
-                    var imagePath = Path.Combine(debugInformationPath, "Screenshot.png");
-                    _context.Scope.Driver.GetScreenshot().SaveAsFile(imagePath);
+                    var imagePath = Path.Combine(debugInformationPath, "FailureScreenshot.png");
+                    _context.TakeScreenshot(imagePath);
 
                     if (_configuration.ReportTeamCityMetadata)
                     {
                         TeamCityMetadataReporter.ReportImage(_testManifest.Name, "Screenshot", imagePath);
                     }
+
+                    var pageScreenshotsPath = Path.Combine(debugInformationPath, "PageScreenshots");
+                    Directory.CreateDirectory(pageScreenshotsPath);
+                    var fileNameFormat = "D" + _pageScreenshots.Count.ToTechnicalString().Length;
```

Comment:
```suggestion
                    var fileNameFormat = "D" + ((int)Math.Log10(_pageScreenshots.Count) + 1);
```

This should be slightly faster and shorter, but more importantly I think it communicates the intention more clearly than a string conversion.

Reply:
I considered this but I've chosen the string length approach specifically because it's clearer to me :D. I'll ask The People.

Reply:
I've added an extension for this, based on Oliver's suggestion.

Reply:
Exquisite